### PR TITLE
Issue #943: 'Bump CS version in REPO' is not working

### LINKED ISSUE #943: 'Bump CS version in REPO' is not working
https://github.com/sevntu-checkstyle/sevntu.checkstyle/actions/runs/3602122111/jobs/6068698973#step:4:16128

```
Error:  Cannot resolve project dependencies:
Error:    Software being installed: eclipsecs-sevntu-plugin 1.43.0
Error:    Missing requirement: eclipsecs-sevntu-plugin 1.43.0 requires 'osgi.bundle; net.sf.eclipsecs.checkstyle 0.0.0' but it could not be found
```

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #943 

---
Action: https://github.com/stoyanK7/sevntu.checkstyle/actions/runs/3613769860/jobs/6089744751
GitHub actions commit: https://github.com/stoyanK7/sevntu.checkstyle/commit/3decd875ef84fbf916890c667a88364bbe385108
 
-----

 - [sed documentation on range addresses](https://www.gnu.org/software/sed/manual/html_node/Range-Addresses.html)
 - https://stackoverflow.com/a/22960617/9553927

### CODE REVIEW

**Review by romani — CHANGES_REQUESTED** [review]
I need to look deeper.
Version bump should not try to compile project.

We just need update digits in file and let other CI job do all compilation in proper form.

Any idea why we need compilation for update of number in file?

We can use sed instead if there is something crazy.

**Comment by rnveach** [comment]
@romani see https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/943#issuecomment-1336203524 . That is the reason.

Is there any chance of asking eclipsecs to be in maven?

**Comment by romani** [comment]
Even dependabot service never compile target project. It simply update numbers in text file. We need same.

**Comment by rnveach** [comment]
> Even dependabot service never compile target project. 

I don't believe we are compiling, but doing version verification. I am not sure, this is all eclipse's process kicking off here.

dependatbot is not enabled for this POM because of this reason. I do not know if dependabot would have the same issues.

Either way, if you want to not do this then we will have to rely on manually parsing.

**Comment by stoyanK7** [comment]
Done.

Github actions commit: https://github.com/stoyanK7/sevntu.checkstyle/commit/825f0ad68900ea67cebce76ff2df1e66ded2a8b9
Job: https://github.com/stoyanK7/sevntu.checkstyle/actions/runs/3622733299/jobs/6107797231

File: .ci/bump-cs-version-in-esclipsecs-plugin.sh [hunk 1]
```diff
@@ -10,6 +10,6 @@ fi
 VERSION=$1
 cd eclipsecs-sevntu-plugin
 
-mvn -e --no-transfer-progress versions:set-property -DgenerateBackupPoms=false -Dproperty=checkstyle.version -DnewVersion="$VERSION"
+sed -i -e "/<properties>/,/<\/properties>/ s|<checkstyle.version>.*</checkstyle.version>|<checkstyle.version>$VERSION</checkstyle.version>|g" pom.xml
```
Comment (romani) [inline .ci/bump-cs-version-in-esclipsecs-plugin.sh:13]: What for this block `<properties>/,/<\/properties>` ?

Comment (stoyanK7) [reply .ci/bump-cs-version-in-esclipsecs-plugin.sh:13]: > Is it a fancy syntax to explain sed to do change only between such tags ?

Correct. 

 - [sed documentation on range addresses](https://www.gnu.org/software/sed/manual/html_node/Range-Addresses.html)
 - https://stackoverflow.com/a/22960617/9553927

[the remark this answers is not in the document]
Comment (stoyanK7) [reply .ci/bump-cs-version-in-esclipsecs-plugin.sh]: Done.
```
✔ ~/Open Source/sevntu.checkstyle [issue/943|✔] 
19:51 $ git diff
✔ ~/Open Source/sevntu.checkstyle [issue/943|✔] 
19:51 $ ./.ci/bump-cs-version-in-eclipsecs-plugin.sh 10.5.1
Version updated to 10.5.1 at eclipsecs-sevntu-plugin/pom.xml
✔ ~/Open Source/sevntu.checkstyle [issue/943|✚ 1] 
19:51 $ git diff
diff --git a/eclipsecs-sevntu-plugin/pom.xml b/eclipsecs-sevntu-plugin/pom.xml
index b1a3911a..68d337e0 100644
--- a/eclipsecs-sevntu-plugin/pom.xml
+++ b/eclipsecs-sevntu-plugin/pom.xml
@@ -28,7 +28,7 @@
   <properties>
     <eclipsecs.version>10.4.0-SNAPSHOT</eclipsecs.version>
     <!-- verify time version -->
-    <checkstyle.version>10.4</checkstyle.version>
+    <checkstyle.version>10.5.1</checkstyle.version>
     <checkstyle.configLocation>https://raw.githubusercontent.com/checkstyle/checkstyle/checkstyle-${checkstyle.version}/config/checkstyle_checks.xml</checkstyle.configLocation>
     <checkstyle.header>https://raw.githubusercontent.com/checkstyle/checkstyle/master/config/java.header</checkstyle.header>
     <checkstyle.regexp.header>https://raw.githubusercontent.com/checkstyle/checkstyle/master/config/java_regexp.header</checkstyle.regexp.header>

```

**Review by romani — CHANGES_REQUESTED** [review]
Ok to merge

**Review by rnveach — APPROVED** [review]

**Pull request merged**